### PR TITLE
Add the posibility to build dbcsr shared library

### PR DESCRIPTION
**Pull request by mtaillefumier**
trivial change needed when the shared version of libcp2k is needed. OFF by default

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -157,7 +157,8 @@ endif ()
 add_library(dbcsr ${DBCSR_SRCS})
 
 set_target_properties(dbcsr PROPERTIES VERSION ${dbcsr_VERSION}
-                                       SOVERSION ${dbcsr_APIVERSION})
+                                       SOVERSION ${dbcsr_APIVERSION}
+			               POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS})
```
Comment (dev-zero) [inline src/CMakeLists.txt:161]: This is only required if `dbcsr` were a `STATIC` library, which it isn't.

File: CMakeLists.txt [hunk 2]
```diff
@@ -69,6 +69,7 @@ set(dbcsr_APIVERSION ${VERSION_MAJOR}.${VERSION_MINOR})
 # OPTIONS
 include(CMakeDependentOption)
 
+option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
```
Comment (dev-zero) [inline CMakeLists.txt:72]: ACK

**Review by dev-zero — CHANGES_REQUESTED** [review]

File: src/CMakeLists.txt (hunk 1 above)
Comment (mtaillefumier) [reply src/CMakeLists.txt]: It does not matter actually. Static libraries can also be build with -fPIC (usually even suggested when we mix static and dynamic libraries). I changed it `POSITION_INDEPENDENT_CODE ON` all the time.

Comment (hfp) [reply src/CMakeLists.txt]: > It does not matter actually. Static libraries can also be build with -fPIC (usually even suggested when we mix static and dynamic libraries). I changed it `POSITION_INDEPENDENT_CODE ON` all the time.

Indeed, very advisable to use Position Independent Code even for all objects even when collected into an archive aka static library. Nowadays, the reason things work without `-fPIC` is that some toolchains apply it by default, e.g., one-step-builds (compile and link into one command). Using `-fPIC` everywhere also technically allows to use the same objects twice (for archive and shared library).

**Comment by hfp** [comment]
@dev-zero Are all requests for change addressed?  If so, let's merge this contribution.

**Comment by dev-zero** [comment]
https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html

So, setting it explicitly does not make sense.

**Comment by mtaillefumier** [comment]
it does as we often link static libraries and dynamic libraries together. Doing so results in potential errors during linking time which is due to the absence of -fPIC during compilation. We can leave it on all the time as the address resolution will happen at linking time if we use archive libraries instead of shared libraries. it has absolutely no influence on performance.

**Review by dev-zero — APPROVED** [review]
Ok, I would really prefer people would use one or the other, rather than mix and match, but that's wishful thinking.

**Pull request merged**